### ISSUE TITLE
Feature Request: Print preview

### ISSUE THREAD

**Issue by johnsie** [notepad-plus-plus/notepad-plus-plus#4236]
### Description of the Issue

Sometimes it's necessary to print the contents of a file, but we have to decide whether it will look ok before potentially wasting paper.

### Steps to Reproduce the Issue

1. File->Print

### Expected Behavior

Either there be a print preview option in the file menu or in the print options

### Actual Behavior

There is no option to see what your print out will look like in advance

### Debug Information

n/a

**Comment by johnsie** [comment]
Known workaround is to print to pdf, but a print preview option would be more user friendly.

**Comment by Mannshoch** [comment]
Print Preview with a slider for Text size (auto-break on), text on Left/Middle/Right and Top/Middle/Down, and possibility to enhance it with Markdown support or similar over Add-on.

**Comment by akoppa** [comment]
No response in more than a year?

**Comment by Mannshoch** [comment]
Seems my last post shocked to much 😁

**Comment by jimp4nse** [comment]
Sadly it seems that this issue will not get priority anytime soon. It has been requested multiple times over the last years. In all discussions, the argument always was that editing documents for later printing would not be the "purpose of np++" (e.g., see https://superuser.com/questions/94290/notepad-display-printing-page-count). This is somewhat hard to understand. There seem to be endless scenarios where printing is desired (monospace font guitar tabs, code examples for academic exercise sheets, simple notes, etc.). Competing editors such as PSPad, having the same "purpose" I guess, support this since forever...

**Comment by sasumner** [comment]
@jimp4nse I think probably the print-to-pdf workaround satisfies most needs these days.  Hard to put effort into something that already has a decent workaround when there are so many other bugs to fix and improvements to make.  My 2c.

**Comment by urheil** [comment]
come on dev! we wont it! pleeeeeeeaaaaaaaaaseeeeeeeeeeeeeeeeeeeeeeeee... hahahaha

is it there an add on or something to do this?

**Comment by softfoot** [comment]
I'd love this option too. Sometimes I need to see the preview to choose landscape or portrait or to trim the text to fit without wrapping.
Dave

**Comment by johnsie** [comment]
There are 1.2k issues with this app. Looks like the dev team have a bit of a backlog. Doubt the will have the capacity to implment this any time soon. It has been over two years now. There are a few nice text editors out there already which do support this feature and better csv handling, so it might be a good idea to shop around.

**Comment by visuttaja** [comment]
Yes. it should be there to avoid distraction on workflow. Some algo snippets needs to be printed . Hands down best print preview implementation is in an old Firefox browser. It could be hacked from there into Notepad++. It could be seen as new improvement. I don't know about new Firefox breed but it might be still there.  It is textbook example how it should be done.
Because of certain plugins I've been forced to use the old Firefox. I'm using IntelliJ for writing Java and it also lacks printing preview.  My first option would be to paste into Notepad++ and get the snippet on paper.  But I need to see it fits beautifully.
That is how I want to live my life.  
Workaround: You can use Firefox: get path by copying it from some explorer view(copy as path) and paste the path to Firefox "Open File" dialog. Then cut the filename and press enter this giving you directory view.  Then Firefox gives you very good print preview.
